### ISSUE TITLE
Is it possible for YOLO team to add attention network model into the detection

### ISSUE THREAD

**Issue by frabob2017** [ultralytics/yolov5#9925]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Is it possible for YOLO team in the future to add the attention network into YOLO detection. Since after detect a lesion, the chance of detect the lesion in the next slice is also high in the same area. Could add attention network into it to make the detection more consistent in the consecutive slices.  I am studying your source code, and it needs a lot of time to understand for me to modify your code.

**Comment by UNeedCryDear** [comment]
how about this pr?
https://github.com/ultralytics/yolov5/pull/9750

**Comment by frabob2017** [comment]
> Yes #9750 offers attention modules.
Thank you both. 

May I ask how to add this attention module into the current YOLO program?  It looks it does not provide the detail how to implement it.

**Comment by frabob2017** [comment]
> Yes #9750 offers attention modules.

I check this module in pytorch "torch.nn.MultiheadAttention", could I add this module in model forward function to make it work?  From #9750,  I saw a lot of models, but I do not know what they are.  I can not find the clear explanation of these modules.

**Comment by frabob2017** [comment]
> 

no one answer this question?

**Comment by UNeedCryDear** [comment]
> May I ask how to add this attention module into the current YOLO program? It looks it does not provide the detail how to implement it.<br>

![image](https://user-images.githubusercontent.com/52729998/198168575-47e85159-0d81-4332-950d-8d8ea11a8776.png)
Add it by your model.yaml like the demo yolov5s-CBAM.YAML
At the same time, the source code and the demo are shown to you. Why do you say there are no implementation details?
> I check this module in pytorch "torch.nn.MultiheadAttention", could I add this module in model forward function to make it work?<br>

No problem in principle, but you need to do it yourself.
> I saw a lot of models, but I do not know what they are. I can not find the clear explanation of these modules.<br>

Please google to search these modules. No one can explain its principle briefly here.

**Comment by frabob2017** [comment]
> > May I ask how to add this attention module into the current YOLO program? It looks it does not provide the detail how to implement it.
> 
> ![image](https://user-images.githubusercontent.com/52729998/198168575-47e85159-0d81-4332-950d-8d8ea11a8776.png) Add it by your model.yaml like the demo yolov5s-CBAM.YAML At the same time, the source code and the demo are shown to you. Why do you say there are no implementation details?
> 
> > I check this module in pytorch "torch.nn.MultiheadAttention", could I add this module in model forward function to make it work?
> 
> No problem in principle, but you need to do it yourself.
> 
> > I saw a lot of models, but I do not know what they are. I can not find the clear explanation of these modules.
> 
> Please google to search these modules. No one can explain its principle briefly here.

Thank you so much. 

Thank you so much. Sorry about that what I comment. I am still learning and may ask some very simple questions because this level is a little beyond my understanding.  My purpose is to reinforce the detection accuracy in the continuous slices. If a lesion is detected in one slice with high confidence, the chance to detect it in the next slice is also very high. I want to use this information as an attention for the next slice to improve the detection accuracy.  I heard the term attention network recently. based on my understanding from YouTube lecture, the attention network will work in this scenario because the continuous slices is a vector of sequence (hope my understanding is correct). Attention network is originally designed for sequence,   I will study it according to your suggestion. Thank you so much.

**Comment by glenn-jocher** [comment]
@frabob2017 you're very welcome! Your learning process is valuable, and your questions are always welcome here. Your understanding is correct, and leveraging the attention network for continuous slice detection sounds like an excellent approach. We're glad to assist you in your endeavors, and we appreciate your thoughtful engagement. Good luck with your studies, and feel free to reach out if you have further questions.
